fast/forms/select/mac-wk2/inactive-appearance.html doesn't produce results

JonWBedard opened the pull request:
#### 03028cddf0c1941a7b591b6320eaf97ad46217d9
<pre>
fast/forms/select/mac-wk2/inactive-appearance.html doesn&apos;t produce results
<a href="https://bugs.webkit.org/show_bug.cgi?id=273441">https://bugs.webkit.org/show_bug.cgi?id=273441</a>
<a href="https://rdar.apple.com/127258496">rdar://127258496</a>

Reviewed by Aditya Keerthi.

* LayoutTests/fast/forms/select/mac-wk2/inactive-appearance.html: Replace js-test.js with
testRunner.waitUntilDone/testRunner.notifyDone.
* LayoutTests/platform/mac-wk2/TestExpectations: fast/forms/select/mac-wk2/inactive-appearance.html
now fails.

Canonical link: <a href="https://commits.webkit.org/278267@main">https://commits.webkit.org/278267@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36d1938ab7e21422b65885cb8ed64a66c999967a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50051 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/29340 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/55/builds/2334 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/53294 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/35461 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/53294 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/52149 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/35461 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/55/builds/2334 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/53294 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/35461 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/55/builds/2334 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe-skia~~](https://ews-build.webkit.org/#/builders/52/builds/8421 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/35461 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/2334 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/54876 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/25145 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/54876 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/26404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/55/builds/2334 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/54876 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/27267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7219 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26132 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/fast/forms/select/mac-wk2/inactive-appearance.html:
```diff
@@ -1,6 +1,7 @@
 <!DOCTYPE html>
 <html>
     <head>
+        <script>window.enablePixelTesting = true;</script>
```
Instead of doing this, what if we dropped `js-test.js` here, and used `testRunner.waitUntilDone()` / `testRunner.notifyDone()`?

Basically:

`jsTestIsAsync = true`;

=>

```
if (window.testRunner)
        testRunner.waitUntilDone();
```

and `finishJSTest();`

=>

```
if (window.testRunner)
        testRunner.notifyDone();
```

JonWBedard replied:
Made that change!

pxlcoder commented on LayoutTests/fast/forms/select/mac-wk2/inactive-appearance.html:
```diff
@@ -11,14 +10,15 @@
 
     await UIHelper.renderingUpdate();
-    finishJSTest();
+    testRunner.notifyDone();
```
```suggestion
    if (window.testRunner)
        testRunner.notifyDone();
```

(just to avoid an exception when running the test manually without WKTR)

JonWBedard committed:
fast/forms/select/mac-wk2/inactive-appearance.html doesn't produce results
https://bugs.webkit.org/show_bug.cgi?id=273441
rdar://127258496

Reviewed by Aditya Keerthi.

* LayoutTests/fast/forms/select/mac-wk2/inactive-appearance.html: Replace js-test.js with
testRunner.waitUntilDone/testRunner.notifyDone.
* LayoutTests/platform/mac-wk2/TestExpectations: fast/forms/select/mac-wk2/inactive-appearance.html
now fails.

Canonical link: https://commits.webkit.org/278267@main
